src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -3,20 +3,45 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.machine.Machine;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
 public class MachineDaoImpl extends AbstractDao implements MachineDao {
-    protected MachineDaoImpl(SessionFactory sessionFactory) {
+    public MachineDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            Transaction transaction = session.beginTransaction();
+            session.save(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
-    public List<Machine> findByAgeOlderThan(int age) {
-        return null;
```

Comment:
I think we need to keep 'age' here and calculate the machine's age based on the year it was made, like this: _2020  - yearMade = age_

Reply:
agree, I guess  you know how to get the current year, right?

Reply:
Done